src/spider/worker/worker.cpp

```diff
@@ -188,6 +188,42 @@ auto get_args_buffers(spider::core::Task const& task)
     return args_buffers;
 }
 
+auto get_args_buffers(
+        std::shared_ptr<spider::core::StorageFactory> const& storage_factory,
+        std::shared_ptr<spider::core::MetadataStorage> const& metadata_store,
+        spider::core::TaskInstance const& instance,
+        spider::core::Task& task
+) -> std::optional<std::vector<msgpack::sbuffer>> {
```

Comment:
Our guidelines specify we should separate the declaration and definition of file-scope functions.
(https://www.notion.so/yscope/WIP-Coding-Guidelines-9a308b847a5343958ba3cb97a850be66?pvs=4#13604e4d9e6b8082aaeaff34664edac8)

However, since this issue extends beyond the scope of the PR lets do it in another PR.

@coderabbitai Could you please open an issue to refactor the file-scope functions to have separate declarations and definitions.